opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java

```diff
@@ -36,92 +36,55 @@
 
 /**
- * Test the SCCSget class.
+ * Test the {@link SCCSget} class.
  * @author Trond Norbye
  */
-public class SCCSgetTest {
+@EnabledForRepository(SCCS)
+class SCCSgetTest {
 
-    private static boolean haveSccs = true;
-    private File sccsfile;
-    private File sccsdir;
-
-    @BeforeAll
-    public static void setUpClass() throws Exception {
-        // Check to see if we have sccs..
-        Process p = null;
-        try {
-            p = Runtime.getRuntime().exec("sccs help help");
-            p.waitFor();
-            haveSccs = (p.exitValue() == 0);
-        } catch (Exception e) {
-            haveSccs = false;
-        } finally {
-            try {
-                if (p != null) {
-                    p.destroy();
-                }
-            } catch (Exception e) {
-
-            }
-        }
-        try {
-            p = Runtime.getRuntime().exec("sccs --version");
-            p.waitFor();
-            haveSccs = (p.exitValue() == 0);
-        } catch (Exception e) {
-            haveSccs = false;
-        } finally {
-            try {
-                if (p != null) {
-                    p.destroy();
-                }
-            } catch (Exception e) {
-
-            }
-        }
-    }
+    private File sccs_file;
+    private File sccs_dir;
 
     @BeforeEach
     public void setUp() throws IOException {
-        if (!haveSccs) {
-            return;
-        }
         try {
-            sccsdir = File.createTempFile("s.test", "sccs");
-            sccsdir.delete();
-            assertTrue(sccsdir.mkdirs(), "Failed to set up the test-directory");
-            sccsfile = new File(sccsdir, "s.note.txt");
-            InputStream in = getClass().getResourceAsStream("/history/s.note.txt");
-            FileOutputStream out = new FileOutputStream(sccsfile);
-            byte[] buffer = new byte[8192];
-            int nr;
-
-            while ((nr = in.read(buffer, 0, buffer.length)) != -1) {
-                out.write(buffer, 0, nr);
+            sccs_dir = File.createTempFile("s.test", "sccs");
+            assertTrue(sccs_dir.delete());
+            assertTrue(sccs_dir.mkdirs(), "Failed to set up the test-directory");
+            sccs_file = new File(sccs_dir, "s.note.txt");
+            try (InputStream in = getClass().getResourceAsStream("/history/s.note.txt");
+                 FileOutputStream out = new FileOutputStream(sccs_file)) {
+
+                byte[] buffer = new byte[8192];
+                int nr;
+
+                while ((nr = in.read(buffer, 0, buffer.length)) != -1) {
+                    out.write(buffer, 0, nr);
+                }
+                out.flush();
             }
-            out.flush();
-            in.close();
-            out.close();
         } catch (IOException ex) {
-            if (sccsfile != null) {
-                sccsfile.delete();
-                sccsdir.delete();
+            if (sccs_file != null) {
```

Comment:
Isn't `@AfterEach` executed even when the test throws exception? In that case this part shouldn't be necessary.

Reply:
good point !